wasm/js/worker.js

```diff
@@ -0,0 +1,31 @@
+import wasm from "../Cargo.toml";
+
+async function initializeWorker(wasm) {
```

Comment:
How does this one differ from the one defined inline in the Rust file below?

Reply:
The `initializeWorker` Rust function accepts a pointer to the Rayon thread. It dereferences that pointer, and then runs the Rayon thread.

The `spawnWorker` JS function spawns the `wasm/js/worker.js` file in a new Worker, and then uses `postMessage` to send the pointer/memory/module:

https://github.com/AleoHQ/sdk/blob/eb8ef4ac36320015088256290457fab502246342/wasm/src/thread_pool/mod.rs#L38-L42

The `initializeWorker` JS function waits for the pointer/memory/module to be received:

https://github.com/AleoHQ/sdk/blob/eb8ef4ac36320015088256290457fab502246342/wasm/js/worker.js#L15-L18

And then after it receives those things it runs the Wasm module inside of the Worker, and calls the Rust `initializeWorker` with the pointer:

https://github.com/AleoHQ/sdk/blob/eb8ef4ac36320015088256290457fab502246342/wasm/js/worker.js#L26

So the Rust `initializeWorker` only runs the Rayon thread, whereas the JS `initializeWorker` manages the Worker, runs the Wasm, and then runs the Rayon thread.

Because it needs to run the Wasm, `wasm/js/worker.js` must be written in JS, it can't be written in Rust.